File: backend/src/main/java/com/staccato/staccato/domain/Feeling.java
package com.staccato.staccato.domain;

import java.util.Arrays;

import com.staccato.exception.StaccatoException;

public enum Feeling {
    HAPPY("happy"),
    ANGRY("angry"),
    SAD("sad"),
    SCARED("scared"),
    EXCITED("excited"),
    NOTHING("nothing");

    private final String feeling;

    Feeling(String feeling) {
        this.feeling = feeling;
    }

    public static Feeling match(String value) {
        return Arrays.stream(values())
                .filter(mood -> mood.getValue().equals(value))
                .findFirst()
                .orElseThrow(() -> new StaccatoException("요청하신 기분 표현을 찾을 수 없어요."));
    }

    public String getValue() {
        return feeling;
    }
}

File: android/Staccato_AN/domain/src/main/java/com/on/staccato/domain/model/Feeling.kt
package com.on.staccato.domain.model

enum class Feeling(val value: String) {
    HAPPY("happy"),
    ANGRY("angry"),
    SAD("sad"),
    SCARED("scared"),
    EXCITED("excited"),
    NOTHING("nothing"),
    ;

    companion object {
        fun fromValue(value: String): Feeling =
            when (value) {
                HAPPY.value -> HAPPY
                ANGRY.value -> ANGRY
                SAD.value -> SAD
                SCARED.value -> SCARED
                EXCITED.value -> EXCITED
                else -> NOTHING
            }
    }
}
